connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/DeadLetterQueueReporter.java

```diff
@@ -119,38 +117,46 @@ public static DeadLetterQueueReporter createAndSetup(Map<String, Object> adminPr
      * @param context processing context containing the raw record at {@link ProcessingContext#consumerRecord()}.
      */
     public void report(ProcessingContext context) {
-        final String dlqTopicName = connConfig.dlqTopicName();
+        Callback callback = (metadata, exception) -> {
+            if (exception != null) {
+                log.error("Could not produce message to dead letter queue. topic=" + dlqTopicName, exception);
+                errorHandlingMetrics.recordDeadLetterQueueProduceFailed();
+            }
+        };
+        report(context, callback);
+    }
+
+    /**
+     * Write the raw records into a Kafka topic. This methods allows for a custom callback to be
+     * passed to the producer.
+     *
+     * @param context processing context containing the raw record at {@link ProcessingContext#consumerRecord()}.
+     * @param callback callback to be invoked by the producer when the record is sent to Kafka.
+     * @return
```

Comment:
Missing JavaDoc details:
```suggestion
     * @return the future associated with the writing of this record; never null
```

Reply:
Why do we need to overload this method to pass in a callback? The only place we're using this new method is via the reporter, and `WorkerErrantRecordReporter.callback` doesn't seem to provide any value and in fact is not able to call `errorHandlingMetrics.recordDeadLetterQueueProduceFailed()` like this class.

Wouldn't it be much simpler to just add a return type to the existing method?